crypto/packet.c

```diff
@@ -510,3 +527,81 @@ void WPACKET_cleanup(WPACKET *pkt)
     }
     pkt->subs = NULL;
 }
+
+int WPACKET_start_quic_sub_packet_bound(WPACKET *pkt, size_t max_len)
+{
+    if (WPACKET_start_sub_packet_len__(pkt,
+                                       ossl_quic_vlint_encode_len(max_len)) == 0)
+        return 0;
+
+    pkt->subs->flags |= WPACKET_FLAGS_QUIC_VLINT;
+    return 1;
+}
+
+int WPACKET_start_quic_sub_packet(WPACKET *pkt)
+{
+    /*
+     * Assume no (sub)packet will exceed 4GiB, thus the 8-byte encoding need not
+     * be used.
+     * */
+    return WPACKET_start_quic_sub_packet_bound(pkt, OSSL_QUIC_VLINT_4B_MIN);
+}
+
+int WPACKET_quic_sub_allocate_bytes(WPACKET *pkt, size_t len, unsigned char **allocbytes)
+{
+    if (   !WPACKET_start_quic_sub_packet_bound(pkt, len)
+        || !WPACKET_allocate_bytes(pkt, len, allocbytes)
+        || !WPACKET_close(pkt))
+        return 0;
+
+    return 1;
+}
+
+/*
+ * Write a QUIC variable-length integer to the packet.
+ */
+int WPACKET_quic_write_vlint(WPACKET *pkt, uint64_t v)
+{
+    unsigned char *b = NULL;
+    size_t enclen = ossl_quic_vlint_encode_len(v);
+
+    if (WPACKET_allocate_bytes(pkt, enclen, &b) == 0)
+        return 0;
+
+    ossl_quic_vlint_encode(b, v);
+    return 1;
+}
+
+int WPACKET_quic_allocate_transport_param_bytes(WPACKET *pkt, uint64_t id,
```

Comment:
These transport_param functions seem oddly specific to a particular part of QUIC. Do we expect the writing of transport params to be sufficiently widespread to justify having specific support for them? If so, why do we not see equivalent functions in the PACKET API?

Reply:
Makes sense. I'll remove these.